BrainPortal/cbrain_plugins/cbrain-plugins-base/cbrain_task/start_vm/bourreau/start_vm.rb

```diff
@@ -0,0 +1,245 @@
+
+#
+# CBRAIN Project
+#
+# Copyright (C) 2008-2012
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.  
+#
+
+require 'net/ssh'
+
+# A task that starts a VM, monitors its booting process using ssh, and
+# mounts the Bourreau task and data directories in it using sshfs.
+class CbrainTask::StartVM < ClusterTask
+
+  Revision_info = CbrainFileRevision[__FILE__]
+
+  # A Hook used to follow the boot process of the VM once it is
+  # running.
+  after_status_transition '*', 'On CPU', :starting
+
+  def setup #:nodoc:
+    errors = param_validation_errors
+    return true if errors.empty?
+    errors.each_key do |key|
+      addlog(errors[key])
+    end
+    return false
+  end
+  
+  def cluster_commands #:nodoc:
+    [ "echo This will never execute" ]  # If the cluster commands are
```

Comment:
Ahhh, ok. Maybe say explicitly that this is precisely want you want to avoid, in the code ?

Does that mean that a dummy bash script will be created and submitted to the cluser with just a echo statement in it?